File: android/src/main/java/com/jasonasmk/clarityx/ClarityPluginX.java
package com.jasonasmk.clarityx;

import androidx.appcompat.app.AppCompatActivity;

import com.microsoft.clarity.Clarity;
import com.microsoft.clarity.ClarityConfig;
import com.microsoft.clarity.models.LogLevel;


public class ClarityPluginX {
    private AppCompatActivity activity;
    public ClarityPluginX(AppCompatActivity activity) {
        this.activity = activity;
    }

    public Boolean initialize(String projectId) {
        ClarityConfig config = new ClarityConfig(projectId);
        config.setLogLevel(LogLevel.Verbose);
        return Clarity.initialize(this.activity, config);
    }

    public Boolean setCustomUserId(String customUserId) {
        return Clarity.setCustomUserId(customUserId);
    }
}

File: ios/Sources/ClarityPluginXPlugin/ClarityPluginX.swift
import Foundation
import Clarity

@objc public class ClarityPluginX: NSObject {
    @objc public func initialize(_ projectId: String) -> Bool {
        let config = ClarityConfig(projectId: projectId)
        return ClaritySDK.initialize(config: config)
    }
    
    @objc public func setCustomUserId(_ customUserId: String) -> Bool {
        return ClaritySDK.setCustomUserId(customUserId)
    }
}
